### ISSUE TITLE
Chapter 5 Spring cloud config server native configuration is not working.

### ISSUE THREAD

**Issue by wangzt568** [ihuaylupo/manning-smia#14]
Error description: If you are using the git profile, you need to set a Git URI in your configuration.  If you are using a native profile and have spring.cloud.config.server.bootstrap=true, you need to use a composite configuration. I have tried other ways like composite. But still not working. Could you help me with this issue?

**Comment by Will-Estes** [comment]
Add to your POM. This fixed mine.

```
		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-starter-bootstrap</artifactId>
		</dependency>
```

**Comment by hannah23280** [comment]
> Add to your POM. This fixed mine.
> 
> ```
> 		<dependency>
> 			<groupId>org.springframework.cloud</groupId>
> 			<artifactId>spring-cloud-starter-bootstrap</artifactId>
> 		</dependency>
> ```

Hi,  i encounter same issue, but have not yet try your solution.  Any idea why adding spring-cloud-starter-bootstrap would solve the problem?  I read online articles, but none mention about using bootstrap

**Comment by hannah23280** [comment]
> > Add to your POM. This fixed mine.
> > ```
> > 		<dependency>
> > 			<groupId>org.springframework.cloud</groupId>
> > 			<artifactId>spring-cloud-starter-bootstrap</artifactId>
> > 		</dependency>
> > ```
> 
> Hi, i encounter same issue, but have not yet try your solution. Any idea why adding spring-cloud-starter-bootstrap would solve the problem? I read online articles, but none mention about using bootstrap

Think i know why, it is needed for bootstrap.yml to work

**Comment by Will-Estes** [comment]
@hanct That's correct, I think the spring team removed the bootstrap dependency from the starter dependencies. So you need to include it separately now.

**Comment by JoroIlchev** [comment]
> @hanct That's correct, I think the spring team removed the bootstrap dependency from the starter dependencies. So you need to include it separately now.

Any idea why when i add this dependency in my pom, now i`m getting this exception:
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.springframework.cloud:spring-cloud-starter-bootstrap:jar is missing. @ line 43, column 15

**Comment by duydoanx** [comment]
You can see the answer here: https://stackoverflow.com/a/65009480

**Comment by AnshuSoni** [comment]
For me, adding , `spring-cloud-starter-bootstrap` dependency did not work, but when I have put same configuration in `application.properties`, it started working. Is it fine ?

**Comment by Will-Estes** [comment]
> For me, adding , `spring-cloud-starter-bootstrap` dependency did not work, but when I have put same configuration in `application.properties`, it started working. Is it fine ?

Since bootstrap has been deprecated for newer versions of Spring, I think putting it in the application.properties is fine. Check out this [StackOverflow ](https://stackoverflow.com/a/65009480)(as posted earlier by duydoanx)

**Comment by Alphabtw00** [comment]
> > For me, adding , `spring-cloud-starter-bootstrap` dependency did not work, but when I have put same configuration in `application.properties`, it started working. Is it fine ?
> 
> Since bootstrap has been deprecated for newer versions of Spring, I think putting it in the application.properties is fine. Check out this [StackOverflow ](https://stackoverflow.com/a/65009480)(as posted earlier by duydoanx)

is it? im still using it and without that dependency bootstrap.properties wont work but with it it works. using spring boot 3.1.4
